src/lib/components/composites/input/Input.svelte

```diff
@@ -0,0 +1,139 @@
+<script lang="ts">
+    import { Input } from "$lib/components/primitives/input/index.js";
+    import { Label } from "$lib/components/primitives/label/index.js";
+    import { cn } from "$lib/utils";
+    import type { WithElementRef } from "bits-ui";
+    import type { Snippet } from "svelte";
+    import type {
+        HTMLButtonAttributes,
+        HTMLInputAttributes,
+        HTMLInputTypeAttribute,
+    } from "svelte/elements";
+    import { z } from "zod";
+
+    type Props = WithElementRef<HTMLInputAttributes> & {
+        inputId: string;
+        label: string;
+        placeholder?: string;
+        required: boolean;
+        type: HTMLInputTypeAttribute;
+        link?: { href: string; text: string };
+        inputClass?: string;
+        schema?: z.ZodFirstPartySchemaTypes;
+        onButtonClick?: () => void;
+        buttonContent?: Snippet;
+        buttonProps?: HTMLButtonAttributes;
+    };
+
+    let {
+        inputId,
+        label,
+        placeholder = "",
+        required,
+        type,
+        link,
+        inputClass = "",
+        schema,
+        onButtonClick,
+        buttonContent,
+        buttonProps,
+        value = $bindable(),
+        class: className,
+        ...restProps
+    }: Props = $props();
+
+    let errorMessages = $state<string[]>([]);
+    let isFirstValidation = $state(true);
+
+    /**
+     * Validates the input value against the provided schema and returns whether the input was valid.
+     *
+     * - If no schema is provided, this method always returns true
+     *
+     * @returns {boolean} True if the input value is valid, false otherwise.
+     */
+    export function validate(): boolean {
+        if (!schema) {
+            return true;
+        }
+
+        isFirstValidation = false;
+        const parsedSchema = schema.safeParse(value);
+        errorMessages = parsedSchema.error?.errors.map((error) => error.message) ?? [];
+        return parsedSchema.success;
+    }
+
+    /**
+     * Get the current value of the input.
+     *
+     * Same as binding the `value` prop.
+     */
+    export const getValue = () => value;
+
+    /**
+     * After the initial input was validated, validate the input on every input event.
+     */
+    function onInput() {
+        if (!isFirstValidation) {
+            validate();
```

Comment:
My idea was, that it would bother the user to show them the validation errors before they even had the chance to input any credentials. So now the user can input their credentials and only in the case where their credentials didn't meet our requirements we show them the error messages.

Reply:
I like the comment, makes this code more understandable imo.